### ISSUE TITLE
Button to flag wrong sentences

### ISSUE THREAD

**Issue by danielsjf** [mozilla/voice-web#890]
Both the recording page and the validation page need a button to flag sentences with grammar or other mistakes. This was already mentioned a few times [here](https://discourse.mozilla.org/t/spelling-grammatical-errors-in-the-sentences/26238), in #863 and in #769. Most of them were corrected manually, but a flag button might be more appropriate and user friendly than filing Github issues.

Alternatively, there could be a similar add and validate procedure for the sentences itself. Only validated sentences (2 or more times) are to be presented to users.

**Comment by Gregoor** [comment]
#756 talks about systematizing sentence collection. And I agree that flagging should also be part of that. We'll especially need that going forward with other languages.

**Comment by jdittrich** [comment]
Also [mentioned at discourse](https://discourse.mozilla.org/t/spelling-grammatical-errors-in-the-sentences/26238).

**Comment by jdittrich** [comment]
Is there a particular blocker on this – like the question if this would be useful, UI Design, or backend infrastructure to register the flags?

**Comment by Gregoor** [comment]
No particular blocker, just not our center of attention atm. Building tools for sentence collection and validation is something we'll definitely look into in the coming weeks. Happy to read people's thoughts on what kind of features/interactions they'd expect to see. 🙂

**Comment by danielsjf** [comment]
It might be useful to have an indication of the variety of the sentences in terms of different words. If you type a sentence, a score similar to what most password fields have (yellow orange green) would be a nice to have.

**Comment by MichaelNMaggs** [comment]
If there's nothing blocking this it would be good to push it up the priority list, especially for recording and validation. The need has become more urgent now that we have very large numbers of Wikipedia sentences that didn't go through the Sentence Collector and which would have blocked by the upload filters if they had (https://github.com/mozilla/voice-web/issues/1958)

If a button could be added I can update the draft guidelines for validation to explain when it should be used: https://discourse.mozilla.org/t/discussion-of-new-guidelines-for-recording-validation/36465

**Comment by dabinat** [comment]
Just to respond to something @Gregoor said in #1966:

> We do already track skipped_sentences (and do nothing with it), maybe we could surface that in the same report or just use that

I don't think skipped sentences is a reliable metric here. Users may skip sentences for reasons other than sentence quality, and I know from validation experience that many users still attempt to read  sentences even if they are bad. (I get the impression a lot of users aren't aware of the ability to skip, but that's probably something for a different thread.)

**Comment by 12people** [comment]
It would also be good to be able to have a choice for WHY a sentence is inappropriate, to further clarify which reasons are valid to report. Off the top of my head:

1. typo
2. grammar (e.g. #769)
3. unclear pronunciation (there are a lot of foreign proper nouns with unclear pronunciations in sentences now, and I think they just muddle up the recordings)

**Comment by MichaelNMaggs** [comment]
@12people  That's on the list, and has I believe been agreed. Not entirely sure exactly when it will appear, though.

**Comment by Gregoor** [comment]
Today is when it'll appear! Reporting is live now. Please let us know if it works as expected.

**Comment by 12people** [comment]
Awesome, it works! :)
